src/v/storage/segment.h

```diff
@@ -318,7 +323,7 @@ inline batch_cache_index::read_result segment::cache_get(
     };
 }
 inline void segment::cache_put(const model::record_batch& batch) {
-    if (likely(bool(_cache))) {
+    if (likely(_cache && _caching_policy != caching_policy::none)) {
```

Comment:
slightly confused about the policy here: is `_cache` not being set the same thing as `caching_policy::none`, and if so, do we need the `none` policy as opposed to not setting the batch cache `_cache` reference?

EDIT: I was thinking that already had a log config option about disabling cache.